lib/Wikia/src/Service/User/Permissions/PermissionsDefinitionImpl.php

```diff
@@ -0,0 +1,414 @@
+	private $groupsAddableByGroup = [];
+
+	/** @var string[string][] - key is the group and value array of groups removable by this group */
+	private $groupsRemovableByGroup = [];
+
+	/** @var string[string][] - key is the group and value array of groups self addable */
+	private $groupsSelfAddableByGroup = [];
+
+	/** @var string[string][] - key is the group and value array of groups self removable */
+	private $groupsSelfRemovableByGroup = [];
+
+	private $globalGroups = [
+		'content-reviewer',
+		'staff',
+		'helper',
+		'vstf',
+		'beta',
+		'bot-global',
+		'util',
+		'reviewer',
+		'poweruser',
+		'translator',
+		'wikifactory',
+		'restricted-login',
+		'council',
+		'authenticated',
+		'wikiastars',
+		'restricted-login',
+		'voldev'
+	];
+
+	private $implicitGroups = [
+		'*',
+		'user',
+		'autoconfirmed',
+		'poweruser'
+	];
+
+	private $permissions = [
+		'apihighlimits',
+		'autoconfirmed',
+		'autopatrol',
+		'bigdelete',
+		'block',
+		'blockemail',
+		'bot',
+		'browsearchive',
+		'createaccount',
+		'createpage',
+		'createtalk',
+		'delete',
+		'deletedhistory',
+		'deletedtext',
+		'deleterevision',
+		'edit',
+		'editinterface',
+		'editmyoptions',
+		'editusercssjs', #deprecated
+		'editusercss',
+		'edituserjs',
+		'hideuser',
+		'import',
+		'importupload',
+		'ipblock-exempt',
+		'markbotedits',
+		'mergehistory',
+		'minoredit',
+		'move',
+		'movefile',
+		'move-rootuserpages',
+		'move-subpages',
+		'nominornewtalk',
+		'noratelimit',
+		'override-export-depth',
+		'patrol',
+		'protect',
+		'proxyunbannable',
+		'purge',
+		'read',
+		'reupload',
+		'reupload-shared',
+		'rollback',
+		'sendemail',
+		'siteadmin',
+		'suppressionlog',
+		'suppressredirect',
+		'suppressrevision',
+		'unblockself',
+		'undelete',
+		'unwatchedpages',
+		'upload',
+		'upload_by_url',
+		'userrights',
+		'userrights-interwiki',
+		'writeapi',
+		'canremovemap',
+		'wikiawidget',
+		'wikifactory',
+		'wikifactorymetrics',
+		'dumpsondemand',
+		'wikifeatures',
+		'MultiFileUploader',
+		'allowedtoblank',
+		'batchmove',
+		'linkstoredirects',
+		'mobilesearches',
+		'soapfailures',
+		'moderatesotd',
+		'hiderevision',
+		'oversight',
+		'abusefilter-modify',
+		'abusefilter-log-detail',
+		'abusefilter-view',
+		'abusefilter-log',
+		'abusefilter-private',
+		'abusefilter-modify-restricted',
+		'abusefilter-revert',
+		'abusefilter-view-private',
+		'abusefilter-hidden-log',
+		'abusefilter-hide-log',
+		'override-antispoof',
+		'checkuser',
+		'checkuser-log',
+		'geocode',
+		'nuke',
+		'refreshspecial',
+		'replacetext',
+		'spamregex',
+		'tboverride', 	// Implies tboverride-account
+		'tboverride-account', 	// For account creation
+		'torunblocked',
+		'abusefilter-bypass',
+		'platinum',
+		'sponsored-achievements',
+		'achievements-exempt',
+		'achievements-explicit',
+		'admindashboard',
+		'commentmove',
+		'commentedit',
+		'commentdelete',
+		'becp_user',
+		'blog-comments-toggle',
```

Comment:
was it somehow addressed that this shouldn't be on this list on wikis that don't have blogs enabled (or at least it wasn't on that list but may be changed) ?